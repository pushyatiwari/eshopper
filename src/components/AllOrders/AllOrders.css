.eachOrder{
    display: flex;
    flex-direction: column;
    background-color: rgb(226, 226, 226);
    margin-bottom: 1rem;
    border-radius: 1rem;
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);}
.allOrders{
    margin:  2rem;
}
.basket-item{
    font-size: larger;
    font-family: 'Times New Roman', Times, serif;
    color: rgb(59, 53, 53);
}