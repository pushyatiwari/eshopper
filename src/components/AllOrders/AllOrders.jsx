import React from 'react';
import './AllOrders.css';
import PropTypes from 'prop-types';
import OrderDetails from '../OrderDetails/OrderDetails';
import ItemDetails from '../ItemDetails/ItemDetails';

// orders: [
//     {
//       orderId: 'o1',
//       items: 12,
//       orderDate: '12-12-2020',
//       amount: 560.00,
//       products: [
//         {
//           id: 'b1',
//           name: 'Banana - Robusta',
//           price: 40,
//           count: 1,
//         },

const renderOrders = (orders) => {
  const renderAllOrders = orders.map((order) => (
    <div className="eachOrder" key={order.orderId}>
      <OrderDetails orders={order} />
      <ItemDetails items={order} />
      <p />
    </div>
  ));
  return renderAllOrders;
};

const AllOrders = ({ orders }) => {
  const allOrders = renderOrders(orders);
  return (
    <div className="allOrders">
      <p className="basket-item">
        ALL ORDERS
      </p>
      <p className="basket-item">
        Past Orders(
        {orders.length}
        )
      </p>
      { allOrders }
    </div>
  );
};

const configShape = {
  orderId: PropTypes.string,
  items: PropTypes.number,
  orderDate: PropTypes.string,
  amount: PropTypes.number,
  products: PropTypes.any,
};
AllOrders.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.shape(configShape)).isRequired,
};
export default AllOrders;
