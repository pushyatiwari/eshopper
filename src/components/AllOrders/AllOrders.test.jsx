import { render, screen } from '@testing-library/react';
import React from 'react';
import AllOrders from './AllOrders';

describe(AllOrders.name, () => {
  const mockOrders = [{
    items:
            {
              'Fruits & Vegatables': [{
                id: 1, name: 'apple', price: 120, count: 1, category: 'Fruits & Vegatables',
              }],
            },
    id: 1,
    date: 1615122360481,
  },

  ];
  test('should display all orders', () => {
    render(<AllOrders orders={mockOrders} />);
    const navbarElement = screen.getByText('ALL ORDERS');
    expect(navbarElement.tagName).toBe('P');
  });
  test('should display past orders with length', () => {
    render(<AllOrders orders={mockOrders} />);
    const navbarElement = screen.getByText(`Past Orders(${mockOrders.length})`);
    expect(navbarElement.tagName).toBe('P');
  });
  test('should be called with', () => {
    const allOrdersContainer = render(
      <AllOrders orders={mockOrders} />,
    );
    expect(allOrdersContainer).toMatchSnapshot();
  });
});
