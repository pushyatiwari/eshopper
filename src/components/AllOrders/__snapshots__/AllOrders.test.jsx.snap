// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllOrders should be called with 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="allOrders"
      >
        <p
          class="basket-item"
        >
          ALL ORDERS
        </p>
        <p
          class="basket-item"
        >
          Past Orders(
          1
          )
        </p>
        <div
          class="eachOrder"
        >
          <div
            class="order-details-container"
          >
            <div
              class="upper-container-order-details"
            >
              <table>
                <tbody>
                  <tr
                    class="order-items-header"
                  >
                    <td>
                      ORDER
                    </td>
                    <td>
                      ITEMS 
                    </td>
                    <td>
                      DATE
                    </td>
                    <td>
                      AMOUNT
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Order id:
                      1
                    </td>
                    <td>
                      1
                       
                      items
                    </td>
                    <td>
                      Sun Mar 07 2021
                    </td>
                    <td>
                      700
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="upper-container"
            >
              <table>
                <tbody>
                  <tr
                    class="orderitems-header"
                  >
                    <td>
                      ITEM DESCRIPTION
                    </td>
                    <td />
                    <td>
                      UNIT PRICE 
                    </td>
                    <td>
                      QUANTITY
                    </td>
                    <td>
                      SUBTOTAL
                    </td>
                  </tr>
                  <tr
                    class="item-category"
                  >
                    <td>
                      Fruits & Vegatables
                    </td>
                    <td />
                    <td />
                    <td />
                    <td />
                  </tr>
                  <tr
                    class="item-details"
                  >
                    <td>
                      apple
                    </td>
                    <td />
                    <td>
                      Rs.
                      120
                    </td>
                    <td>
                      1
                    </td>
                    <td>
                      Rs.
                      120
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <p />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="allOrders"
    >
      <p
        class="basket-item"
      >
        ALL ORDERS
      </p>
      <p
        class="basket-item"
      >
        Past Orders(
        1
        )
      </p>
      <div
        class="eachOrder"
      >
        <div
          class="order-details-container"
        >
          <div
            class="upper-container-order-details"
          >
            <table>
              <tbody>
                <tr
                  class="order-items-header"
                >
                  <td>
                    ORDER
                  </td>
                  <td>
                    ITEMS 
                  </td>
                  <td>
                    DATE
                  </td>
                  <td>
                    AMOUNT
                  </td>
                </tr>
                <tr>
                  <td>
                    Order id:
                    1
                  </td>
                  <td>
                    1
                     
                    items
                  </td>
                  <td>
                    Sun Mar 07 2021
                  </td>
                  <td>
                    700
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="upper-container"
          >
            <table>
              <tbody>
                <tr
                  class="orderitems-header"
                >
                  <td>
                    ITEM DESCRIPTION
                  </td>
                  <td />
                  <td>
                    UNIT PRICE 
                  </td>
                  <td>
                    QUANTITY
                  </td>
                  <td>
                    SUBTOTAL
                  </td>
                </tr>
                <tr
                  class="item-category"
                >
                  <td>
                    Fruits & Vegatables
                  </td>
                  <td />
                  <td />
                  <td />
                  <td />
                </tr>
                <tr
                  class="item-details"
                >
                  <td>
                    apple
                  </td>
                  <td />
                  <td>
                    Rs.
                    120
                  </td>
                  <td>
                    1
                  </td>
                  <td>
                    Rs.
                    120
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <p />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
